### PR TITLE
Move TestPlan UI server out of `WebServerExporter`

### PR DESCRIPTION
**Pull request by raoyitao**
* The `WebServerExporter` can generate a JSON file and start a web
  server at background so that the JSON report can be opened in a
  browser, however there are problems if this implementation.
  - The JSON file is generated at default location and may overwrite
    another file with same name, should provide a new config option
    to specify the path of JSON report.
  - The tpr script calls `WebServerExporter` to display a report in
    Web UI, A common case is that when displaying "./report.json"
    in current directory the json file will be overwritten (although
    JSON content can be kept same but modified time of file is changed,
    that's side effect). And if there are attachments they will also
    be copied, which leads to exception.
  The Web UI server can be standalone module and can be used by tpr
  script as well as used by `WebServerExporter`.
* Some code optimization.

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [ ] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/cli/commands/writers.py [hunk 1]
```diff
@@ -84,10 +91,21 @@ def __init__(self, port: int):
...
 
-        exporter = WebServerExporter(ui_port=self.port)
-        exporter.export(result)
-        exporter.wait_for_kb_interrupt()
+        with tempfile.NamedTemporaryFile("w+", suffix=".json") as fp:
+            json.dump(data, fp)
+
+            # Save any attachments.
+            attachments_dir = os.path.join(
+                os.path.dirname(fp.name), defaults.ATTACHMENTS
```
[the remark this answers is not in the document]
Comment (raoyitao) [reply testplan/cli/commands/writers.py:104]: In our implementation attaching a single file or attaching a directory, the destination file will have a unique prefix in "_attachments" directory, a corner case is that advanced user create `Attachment` or `Directory` entries and add it into the report then they can specify the `dst_path`, that would be a conflict. Or create a temp directory and put json report and "_attachments" in that directory, then it is easy to remove all of then (simpler than the current way of removing `moved_attachments` one by one and then removing "_attachment" directory if empty).

And it's reasonable to re-use code of generating json report and attachments, let me think about where to place the common part.

[the remark this answers is not in the document]
Comment (raoyitao) [reply testplan/cli/commands/writers.py]: Fixed by calling `JsonExporter` to generate test report. And for tpr script the temp directory will automatically be removed with a context manager.

**Review by Pyifan — COMMENTED** [review]
Per offline discussion with Yitao, proposed workflow for user is:

tpr convert fromdb $doc_id tojson $doc_id.json
#download from db
#read report into memory
#call json exporter

tpr display fromjson $doc_id.json
**#call json exporter -> tmpdir (can this step be optimized @kn-ms )**
#call web_ui_server with tmpdir

I think overall the workflow looks reasonable to me. Let see the highlighted step can be optimized, in this PR or in a future PR?

File: testplan/cli/commands/writers.py [hunk 2]
```diff
@@ -84,10 +88,16 @@ def __init__(self, port: int):
...
-
-        exporter = WebServerExporter(ui_port=self.port)
-        exporter.export(result)
-        exporter.wait_for_kb_interrupt()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            json_path = os.path.join(tmpdir, "report.json")
+            exporter = JSONExporter(
+                json_path=json_path, split_json_report=False
+            )
+            exporter.export(result)
```
Comment (Pyifan) [inline testplan/cli/commands/writers.py:96]: This part is repeating WebServerExporter logic, I'm fine if we leave a comment saying this part is to be optimized.

**Review by Pyifan — CHANGES_REQUESTED** [review]

**Review by Pyifan — APPROVED** [review]

**Pull request merged**